src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/common/world/MixinPathNavigationRegion.java

```diff
@@ -0,0 +1,152 @@
+//    private ChunkAccess[][] no2DArray(Level level, BlockPos blockPos, BlockPos blockPos2) {
+//        if (!((CubicLevelHeightAccessor) level).isCubic()) {
+//            return new ChunkAccess[][]
+//        }
+//    }
+
+
+    @Redirect(method = "<init>", at = @At(value = "INVOKE", target = "Lnet/minecraft/core/SectionPos;blockToSectionCoord(I)I"))
+    private int zeroOutExistingIterators(int coord, Level world, BlockPos blockPos, BlockPos blockPos2) {
+        if (!((CubicLevelHeightAccessor) world).isCubic()) {
+            return SectionPos.blockToSectionCoord(coord);
+        }
+
+        return 0;
+    }
+
+
+    @Inject(method = "<init>", at = @At("TAIL"))
+    private void useCubes(Level world, BlockPos blockPos, BlockPos blockPos2, CallbackInfo ci) {
+        this.isCubic = ((CubicLevelHeightAccessor) world).isCubic();
+        this.minY = world.getMinBuildHeight();
+        this.height = world.getHeight();
+
+        if (!isCubic) {
+            return;
+        }
+        this.centerX = Coords.blockToCube(blockPos.getX());
+        this.centerY = Coords.blockToCube(blockPos.getY());
+        this.centerZ = Coords.blockToCube(blockPos.getZ());
+
+        int offsetX = Coords.blockToCube(blockPos2.getX());
+        int offsetY = Coords.blockToCube(blockPos2.getY());
+        int offsetZ = Coords.blockToCube(blockPos2.getZ());
+
+        this.cubes = new ChunkAccess[(offsetX - centerX + 1) * (offsetY - centerY + 1) * (offsetZ - centerZ + 1)];
+
+        this.diameter = Mth.floor(Math.cbrt(cubes.length));
+
+        ICubeProvider cubeProvider = (ICubeProvider) world.getChunkSource();
+
+        for (int x = this.centerX; x <= offsetX; ++x) {
+            for (int y = this.centerY; y <= offsetY; ++y) {
+                for (int z = this.centerZ; z <= offsetZ; ++z) {
+                    int dx = x - this.centerX;
+                    int dy = y - this.centerY;
+                    int dz = z - this.centerZ;
+                    int index = diameter * (dx * diameter + dy) + dz;
+                    this.cubes[index] = cubeProvider.getCubeNow(x, y, z);
+                }
+            }
+        }
+
+        // Who thought this was a good idea? BC THIS FIELD LITERALLY DOES NOTHING >:(
+//        for (int x = this.centerX; x < offsetX; x++) {
+//            for (int y = this.centerY; y < offsetY; y++) {
+//                for (int z = this.centerZ; z < offsetZ; z++) {
+//                    int dx = x - this.centerX;
+//                    int dy = y - this.centerY;
+//                    int dz = z - this.centerZ;
+//                    int index = diameter * (dx * diameter + dy) + dz;
+//                    ChunkAccess chunkAccess = this.cubes[index];
+//                    if (chunkAccess != null && !chunkAccess.isYSpaceEmpty(blockPos.getY(), blockPos2.getY())) {
+//                        this.allEmpty = false;
+//                        return;
+//                    }
+//                }
+//            }
+//        }
```

Comment:
Can this be removed?

Reply:
I have it commented out currently because this is technically replicating the vanilla constructor but... that field being set doesnt absolutely nothing in this class or anywhere else for that matter